src/matchers/__tests__/to-have-accessibility-value.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { View } from 'react-native';
+import { render, screen } from '../..';
+import '../extend-expect';
+
+function renderViewsWithAccessibilityValue() {
+  return render(
+    <>
+      <View testID="min" accessibilityValue={{ min: 0 }} />
+      <View testID="max" accessibilityValue={{ max: 100 }} />
+      <View testID="now" accessibilityValue={{ now: 65 }} />
+      <View testID="min-max" accessibilityValue={{ min: 0, max: 100 }} />
+      <View
+        testID="min-max-now"
+        accessibilityValue={{ min: 0, max: 100, now: 65 }}
+      />
+      <View
+        testID="min-max-now-text"
+        accessibilityValue={{ text: 'test', min: 0, max: 100, now: 65 }}
+      />
+      <View
+        testID="text"
+        accessibilityValue={{ text: 'accessibility value' }}
+      />
+    </>
+  );
+}
+
+test('toHaveAccessibilityValue() on matching accessibility value', () => {
```

Comment:
Imo these tests are not very readable for two reasons: 

- the first one is because of the `renderViewsWithAccessibilityValue` util. It reduces code duplication but code duplication in tests is not a big deal and here to understand every test it is necessary to look its definition to understand the test which is not great
- the second is the use of variables in the test, e.g. `min` or `minMax`, for the same reason. The variable names actually have less meaning than how they were queried so I'd rather use queries directly.

The test cases could be organized differently to avoid rendering so many components in each tests. You could basically write the same tests than in the [tests or the role query ](https://github.com/callstack/react-native-testing-library/blob/main/src/queries/__tests__/role-value.test.tsx). Also what is missing in those tests and that's present in the role query tests is the support for `aria props`

Reply:
[nit] Yeah, I understand that the tests resemble what Jest Native had, and I think that @pierrezimmermannbam has a point here, that by grouping the scenarios into more smaller tests the tests would be more readable.